MiniApp/Classes/MiniAppDownloader.swift

```diff
@@ -22,52 +18,70 @@ class MiniAppDownloader: NSObject, MiniAppDownloaderProtocol {
         return operationQueue
     }()
 
-    lazy var session: URLSession = {
-        let configuration = URLSessionConfiguration.default
-        return URLSession(configuration: configuration, delegate: self, delegateQueue: queue)
-    }()
+    init() {
+        self.miniAppStorage = MiniAppStorage()
+        self.manifestDownloader = ManifestDownloader()
+    }
 
-    func download(_ urls: [String], to pathDirectory: URL) {
-        for url in urls {
-            guard let fileDirectory = UrlParser.parseForFileDirectory(with: url),
-                let downloadUrl = URL(string: url)
-            else {
-                #if DEBUG
-                    print("MiniAppSDK: Invalid URL from manifest.")
-                #endif
-                return
+    func download(with apiClient: MiniAppClient, appId: String, versionId: String, completionHandler: @escaping (Result<URL?, Error>) -> Void) {
+        self.miniAppClient = apiClient
+        self.manifestDownloader.fetchManifest(apiClient: apiClient, appId: appId, versionId: versionId) { (result) in
+            switch result {
+            case .success(let responseData):
+                self.completionHandler = completionHandler
```

Comment:
If the completionHandler references the same MiniAppDownloader instance this would cause a retain cycle.

Reply:
Yeah, I felt the same. But I guess this can be handled by creating a state machine for downloading that could help not only to avoid these memory leaks but also to have more control over download process. This PR is for base implementation of downloading and storing the mini app. Definitely these changes will be done when we consider to update them with the state machine. Or may be we can use some 3rd party pods just for networking :wink: